src/progaudi.tarantool/RequestWriter.cs

```diff
@@ -79,56 +85,88 @@ public void Dispose()
         private void WriteFunction()
         {
             var handles = new[] { _exitEvent.WaitHandle, _newRequestsAvailable.WaitHandle };
-
+            var throttle = _connectionOptions.WriteThrottlePeriodInMs;
+            long remaining;
             while (true)
             {
                 switch (WaitHandle.WaitAny(handles))
                 {
                     case 0:
                         return;
                     case 1:
-                        WriteRequests(200);
+                        WriteRequests(_connectionOptions.WriteStreamBufferSize,
+                            _connectionOptions.MaxRequestsInBatch);
+
+                        remaining = Interlocked.Read(ref _remaining);
+
+                        // Thread.Sleep will be called only if the number of pending bytes less than
+                        // MinRequestsWithThrottle
+
+                        if (throttle > 0 && remaining < _connectionOptions.MinRequestsWithThrottle)
+                            Thread.Sleep(throttle);
+
                         break;
                     default:
                         throw new ArgumentOutOfRangeException();
                 }
             }
         }
 
-        private void WriteRequests(int limit)
+        private void WriteRequests(int bufferLength, int limit)
         {
             void WriteBuffer(ArraySegment<byte> buffer)
             {
                 _physicalConnection.Write(buffer.Array, buffer.Offset, buffer.Count);
             }
 
-            Tuple<ArraySegment<byte>, ArraySegment<byte>> GetRequest()
+            bool GetRequest(out ArraySegment<byte> result)
             {
                 lock (_lock)
                 {
                     if (_buffer.Count > 0)
-                        return _buffer.Dequeue();
+                    {
+                        _remaining = _buffer.Count + 1;
+                        result = _buffer.Dequeue();
+                        return true;
+                    }
                 }
 
-                return null;
-            }
+                result = default;
+                return false;
+             }
 
-            Tuple<ArraySegment<byte>, ArraySegment<byte>> request;
+            ArraySegment<byte> request;
             var count = 0;
-            while ((request = GetRequest()) != null)
+            UInt64 length = 0;
```

Comment:
This is the only place where var is not recommended.

Reply:
@karlovnv Do you support Russia?